clap/multiarg.zig

```diff
@@ -0,0 +1,166 @@
+                .None => &flags,
+                .One => &single_options,
+                .Many => &multi_options,
+            };
+            index = ptr.*;
+            ptr.* += 1;
+        }
+
+        const converted = clap.Param(usize){
+            .id = index,
+            .names = param.names,
+            .takes_value = param.takes_value,
+        };
+        converted_params = converted_params ++ [_]clap.Param(usize){converted};
+    }
+
+    return struct {
+        single_options: [single_options]?[]const u8,
+        multi_options: [multi_options]std.ArrayList([]const u8),
+        flags: [flags]bool,
+        pos: []const []const u8,
+        allocator: *mem.Allocator,
+
+        pub fn parse(allocator: *mem.Allocator, comptime ArgIter: type, iter: *ArgIter) !@This() {
+            var pos = std.ArrayList([]const u8).init(allocator);
+            var res = @This(){
+                .single_options = [_]?[]const u8{null} ** single_options,
+                .multi_options = [_]std.ArrayList([]const u8){undefined} ** multi_options,
+                .flags = [_]bool{false} ** flags,
+                .pos = undefined,
+                .allocator = allocator,
+            };
+            for (res.multi_options) |*init_opt| {
+                init_opt.* = std.ArrayList([]const u8).init(allocator);
+            }
+
+            var stream = clap.StreamingClap(usize, ArgIter){
+                .params = converted_params,
+                .iter = iter,
+            };
+            while (try stream.next()) |arg| {
+                const param = arg.param;
+                if (param.names.long == null and param.names.short == null) {
+                    try pos.append(arg.value.?);
+                } else if (param.takes_value == .One) {
+                    debug.assert(res.single_options.len != 0);
+                    res.single_options[param.id] = arg.value.?;
+                } else if (param.takes_value == .Many) {
+                    debug.assert(res.multi_options.len != 0);
+                    try res.multi_options[param.id].append(arg.value.?);
+                } else {
+                    debug.assert(res.flags.len != 0);
+                    res.flags[param.id] = true;
+                }
+            }
+
+            res.pos = pos.toOwnedSlice();
+            return res;
+        }
+
+        pub fn deinit(parser: *@This()) void {
+            for (parser.multi_options) |o|
+                o.deinit();
+            parser.allocator.free(parser.pos);
+            parser.* = undefined;
+        }
+
+        pub fn flag(parser: @This(), comptime name: []const u8) bool {
+            const param = comptime findParam(name);
+            if (param.takes_value != .None)
+                @compileError(name ++ " is an option and not a flag.");
+
+            return parser.flags[param.id];
+        }
+
+        pub fn multiOptions(parser: @This(), comptime name: []const u8) [][]const u8 {
```

Comment:
Maybe just call this `options`? Idk, I like short names and this may just be a small nit.

Reply:
Ah, yes, I did call it that originally, but it was overlapping with the `options` field. Not a problem now that it's been split up.